xarray/tests/test_missing.py

```diff
@@ -439,3 +445,82 @@ def test_ffill_dataset(ds):
 @requires_bottleneck
 def test_bfill_dataset(ds):
     ds.ffill(dim="time")
+
+
+@requires_bottleneck
+@pytest.mark.parametrize(
+    "y, lengths",
+    [
+        [np.arange(9), [[3, 3, 3, 0, 2, 2, 0, 2, 2]]],
+        [np.arange(9) * 3, [[9, 9, 9, 0, 6, 6, 0, 6, 6]]],
+        pytest.param(
+            [0, 2, 5, 6, 7, 8, 10, 12, 14],
+            [[6, 6, 6, 0, 2, 2, 0, 3, 3]],
+            marks=pytest.mark.xfail(
+                reason="max_gap with irregularly spaced coordinate."
```

Comment:
Forgive me if I'm being slow—is the `max_gap` measuring the locations (i.e. on the index), or the number of values? The [example below](https://github.com/pydata/xarray/pull/3302/files#diff-141a2535ec6dc1dd95b48446d8965bdaR525) seems to be counting two values, rather than measuring the space between the locations. 

If that's right, why does it matter than the coordinates are irregular (or even non-monotonic)?